sympy/solvers/ode/single.py

```diff
@@ -1889,5 +1889,92 @@ def _get_general_solution(self, *, simplify_flag: bool = True):
         return fsol
 
 
+class Hypergeometric2nd(SingleODESolver):
+    hint = "2nd_hypergeometric"
+    has_integral = True
+
+    def _matches(self):
+        eq = self.ode_problem.eq_preprocessed
+        func = self.ode_problem.func
+        x = self.ode_problem.sym
+        df = func.diff(x)
+        a3 = Wild('a3', exclude=[func, func.diff(x), func.diff(x, 2)])
+        b3 = Wild('b3', exclude=[func, func.diff(x), func.diff(x, 2)])
+        c3 = Wild('c3', exclude=[func, func.diff(x), func.diff(x, 2)])
+        deq = a3*(func.diff(x, 2)) + b3*df + c3*func
```

Comment:
The code for getting the coefficients of a linear ODE should not just be in this particular solver. Many solvers need it.

Reply:
In `SinglePatternODESolver`, there is `wilds()` method, so should I use pattern matching for this?

Reply:
I think that the ODE problem class should have a method for getting the coefficients of a linear ODE:
https://github.com/sympy/sympy/blob/ce8b52f462ac1017f37bfef944c095a0b5756be0/sympy/solvers/ode/single.py#L144-L148

Reply:
> The hypergeometric.py module needs to have some kind of docstring or comments at the top that explains what it is including a reference for the algorithm and an explanation of which parts are implemented. Also `Hypergeometric2nd` needs a docstring.
> 
> Is this method included in the docs at all?

No, there isn't any docstring related to hypergeometric, there were only few comments in the code to make it readable.